BridgeCareApp/BridgeCareCodeFirst/Models/BudgetReportData.cs

```diff
@@ -0,0 +1,137 @@
+                    Hashtable yearView;
+                    if (!BudgetYearView.Contains(report.Budget))
+                    {
+                        yearView = new Hashtable();
+                        BudgetYearView.Add(report.Budget, yearView);
+                    }
+                    else
+                    {
+                        yearView = (Hashtable)BudgetYearView[report.Budget];
+                    }
+
+                    if (yearView.Contains(report.Years))
+                    {
+                        sum = (double)yearView[report.Years];
+                        yearView.Remove(report.Years);
+                    }
+                    sum += cost;
+                    yearView.Add(report.Years, sum);
+                    costList.Add(new CostDetails
+                    {
+                        Cost = report.Cost_.Value,
+                        Years = report.Years,
+                        Budget = report.Budget
+                    });
+                }
+                foreach (string item in BudgetYearView.Keys)
+                {
+                    if (!budgetTypes.Contains(item))
+                    {
+                        BudgetYearView.Remove(item);
+                    }
+                }
+            }
+            catch (SqlException ex)
+            {
+                db.Dispose();
+                if (ex.Number == -2 || ex.Number == 11)
+                {
+                    throw new TimeoutException("The server has timed out. Please try after some time");
+                }
+                if (ex.Number == 208)
+                {
+                    throw new InvalidOperationException("Network or simulation table does not exist in the database");
+                }
+            }
+            catch (OutOfMemoryException)
+            {
+                db.Dispose();
+                throw new OutOfMemoryException("The server is out of memory. Please try after some time");
+            }
+            db.Dispose();
+            var budgetReportDetails = new BudgetReportDetails
+            {
+                BudgetForYear = BudgetYearView,
+                CostDetails = costList
+            };
+            return budgetReportDetails;
+        }
+
+        public string[] InvestmentData(SimulationResult data)
+        {
+            var db = new BridgeCareContext();
+            var budgetOrder = db.INVESTMENTs.Where(_ => _.SIMULATIONID == data.SimulationId)
+                 .Select(_ => _.BUDGETORDER)
+                 .First();
+            if (budgetOrder == "" || budgetOrder == null)
+            {
+                db.Dispose();
+                throw new Exception("Budget types not found in Investments table for the id : " + data.SimulationId);
+            }
+            var budgetTypes = budgetOrder.Split(',');
+            db.Dispose();
+            return budgetTypes;
+        }
+
+        public class CostDetails : ICostDetails
```

Comment:
Move these to separate classes.

Reply:
Moved all the interfaces to separate folder